oadp-operator/controllers/velero.go

```diff
@@ -1,14 +1,469 @@
 package controllers
 
 import (
+	"fmt"
 	"github.com/go-logr/logr"
 	oadpv1alpha1 "github.com/openshift/oadp-operator/api/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	"k8s.io/utils/pointer"
+)
+
+const (
+	Velero          = "velero"
+	VeleroNamespace = "oadp-operator"
+	//TODO: Check for default secret names
+	VELERO_AWS_SECRET_NAME   = "cloud-credentials"
+	VELERO_AZURE_SECRET_NAME = "cloud-credentials-azure"
+	VELERO_GCP_SECRET_NAME   = "cloud-credentials-gcp"
+)
+
+// Environment Vars
+const (
+	LD_LIBRARY_PATH               = "LD_LIBRARY_PATH"
+	VELERO_NAMESPACE              = "VELERO_NAMESPACE"
+	VELERO_SCRATCH_DIR            = "VELERO_SCRATCH_DIR"
+	AWS_SHARED_CREDENTIALS_FILE   = "AWS_SHARED_CREDENTIALS_FILE"
+	AZURE_SHARED_CREDENTIALS_FILE = "AZURE_SHARED_CREDENTIALS_FILE"
+	GCP_SHARED_CREDENTIALS_FILE   = "GCP_SHARED_CREDENTIALS_FILE"
+	//TODO: Add Proxy env vars
+	HTTP_PROXY  = "HTTP_PROXY"
+	HTTPS_PROXY = "HTTPS_PROXY"
+	NO_PROXY    = "NO_PROXY"
+)
+
+//TODO: Add Image customization options
+// Images
+const (
```

Comment:
NIT: you will need to make the tags and images constants we can override with env vars

Reply:
We can do this in a future update though